pkg/utils/gardener/deletion_confirmation.go

```diff
@@ -56,26 +53,13 @@ func CheckIfDeletionIsConfirmed(obj client.Object) error {
 	return nil
 }
 
-// ConfirmDeletion adds Gardener's deletion confirmation annotation to the given object and sends an UPDATE request.
-func ConfirmDeletion(ctx context.Context, c client.Client, obj client.Object) error {
-	return retry.RetryOnConflict(retry.DefaultBackoff, func() error {
-		if err := c.Get(ctx, client.ObjectKeyFromObject(obj), obj); err != nil {
-			if !errors.IsNotFound(err) {
-				return err
-			}
-			return nil
-		}
-
-		existing := obj.DeepCopyObject()
-		kubernetes.SetMetaDataAnnotation(obj, ConfirmationDeletion, "true")
-		kubernetes.SetMetaDataAnnotation(obj, v1beta1constants.GardenerTimestamp, TimeNow().UTC().String())
-
-		if reflect.DeepEqual(existing, obj) {
-			return nil
-		}
-
-		return c.Update(ctx, obj)
-	})
+// ConfirmDeletion adds Gardener's deletion confirmation and timestamp annotation to the given object and sends a PATCH
+// request. It does not ignore `NotFound` errors while patching.
```

Comment:
This is done to handle `NotFound` errors explicitly, i.e. if we can't confirm the deletion of an object, because it's already gone, we don't have to delete it anymore :) 
E.g. https://github.com/gardener/gardener/pull/3979/commits/c767cc9a3db47a887f67065bcd5a045a8f1ac336#diff-f9dad471fefe46d915e28461c2ea20891c5fe29656999f54f469d58a48717590R151-R153